src/components/InventoryGroupDetail/GroupDetailHeader.js

```diff
@@ -66,6 +66,7 @@ const GroupDetailHeader = ({ groupId }) => {
             await getInventoryGroupDevicesUpdateInfo(groupId);
           setEdgeDeviceUpdateInfo(groupEdgeDevicesUpdateInfo);
         } catch (error) {
+          navigate('/groups');
```

Comment:
I am not sure edge devices update info is the best endpoint to check if inventory group exists or not. It is very tied to the edge devices logic, and this also wouldn't work with edge feature flag set to false. 

When a group is deleted, GET /groups/%id should return an empty response, and this is the best source of knowledge to understand if a given group still exists or not. I suggest to look into https://github.com/RedHatInsights/insights-inventory-frontend/blob/master/src/components/InventoryGroupDetail/InventoryGroupDetail.js#L55: here, in `data`, we receive the response after fetching GET /groups/%id. From this selector, you can also pick the key "fulfilled" and other flags: if the request is fulfilled, but the response has empty list => the group doesn't exist. I think we can combine this two facts and write a new useEffect in this component.